src/esp/sensor/CameraSensor.cpp

```diff
@@ -150,6 +140,11 @@ bool CameraSensor::drawObservation(sim::Simulator& sim) {
   if (sim.isFrustumCullingEnabled())
     flags |= gfx::RenderCamera::Flag::FrustumCulling;
 
+  // Set projection matrix, transformation matrix, and viewport
+  renderCamera_->setProjectionMatrix(width_, height_, projectionMatrix_);
+  renderCamera_->node().setTransformation(this->computeTransformationMatrix());
+  renderCamera_->setViewport(this->framebufferSize());
```

Comment:
You can leave it as this for now.
But the logic is the same as the projection matrix, namely, you **only need to set it once** if there is any changes